src/summary/profile/demographics-card.component.tsx

```diff
@@ -9,30 +8,25 @@ import dayjs from "dayjs";
 import { age } from "./age-helpers";
 
 export default function DemographicsCard(props: DemographicsCardProps) {
-  const [currentPatient, setCurrentPatient] = React.useState(null);
-
-  React.useEffect(() => {
-    const subscription = getCurrentPatient().subscribe(patient =>
-      setCurrentPatient(patient)
-    );
-
-    return () => subscription.unsubscribe();
-  });
-
```

Comment:
Decided to have `profile-section` pass in `currentPatient` instead of repeating this code in every widget.

Reply:
Shouldn’t we did this for all of patient chart?

Reply:
good point

Reply:
I slightly prefer subscribing to the patient in each component -- it doesn't cause a network request so it's totally fine from a perf perspective. A bit easier than plumbing the prop everywhere. Additionally, moving the subscription into the parent component means that whenever the patient re-renders, *all* of the patient chart components are re-rendered even if they don't need to. Whereas each one subscribing individually means that only the components that rely on the patient object re-render.

How about this - we create a hook called `useCurrentPatient` inside of esm-api and export it out for esm-patient-chart and other applications to use? Then in each component, you can just do the following:

```js
import { useCurrentPatient } from '@openmrs/esm-api';

function Foo() {
  const patient = useCurrentPatient();
  return <button>{patient.thing}</button>
}
```

Thoughts?

Reply:
I preferred subscribing to the patient in the parent component because of the following reasons:

1) the patient chart should, in fact, be re-rendered when a patient changes, I don't think there is a situation where we wouldn't want to re-render a patient chart component when loading a different patient, as all the components are dependent on the current patient. Having a single source of truth will make sure all components are displaying data for the same patient.

2) writing tests for these components are much easier, no need to keep mocking the `useCurrentPatient` function.

3) I feel like writing `useCurrentPatient` in each component is repetitive.

Reply:
I disagree, but not enough to block approval/merging. Let's go forward with what you have.